api/agent/call.go

```diff
@@ -392,6 +407,21 @@ func (c *call) End(ctx context.Context, errIn error) error {
 	return errIn // original error, important for use in sync call returns
 }
 
+// getCPUQuota converts CPUs to CPU quota by multiplying them by 100.
+// eg. 0.20 becomes 20, 2.0 becomes 200, which are easier to calculate
+// and manage as resources and translate to Linux CFS (cpu-quota and cpu-period)
+func getCPUQuota(CPUs string) (uint64, error) {
```

Comment:
So, the flow is more like string -> float -> int. I opted for string since we may extend it to k8s style milicpus, etc. formats. We end up converting all of this to uint64 at driver level:

https://godoc.org/github.com/moby/moby/api/types/container#Resources

Maybe we should stick with uint64 from the front and stay consistent using millicpu, where "100" means 100% (1CPU), etc.

Reply:
Assuming we are ok with "string -> float -> int" flow, I'm going to clean up (and clarify this) further in the next commit.

Reply:
it would be nice if it were either a float or an int, and then the type for it can add `UnmarshalJSON`, etc., if we want the format to be `"cpus":"100mCPU"` or something, it would be pretty flexible. i think we should just pick something, though, instead of having a string we have to parse out much later down the line. then the user is confronted with their poor choices when they're adding it to the route and there's no chance of a mishap (error) in the backend, we can just use the thing.

i.e. there'd be no string->float->int, just parse to an int or float immediately, and then the driver can do what it needs to with that. it's a little strange to see a number in a string, as it is, in a json api. I'd be okay with just picking 'milliCPUs' to be consistent with k8s now and calling it a day. if we end up regretting that it should be easy to work around, but I doubt it.